### PR TITLE
refactor: [FX-2180] Make ShowMore BASE 2 compliant

### PR DESCRIPTION
**Pull request by LashaJini**
[FX-2180](https://toptal-core.atlassian.net/browse/FX-2180?atlOrigin=eyJpIjoiM2M1ODhmYzM4MDNhNDkzNjlkNDNkYmY5ZDQ0YTg3Y2YiLCJwIjoiaiJ9)

### Description

- `ShowMore` component should have action button  with an `iconPosition` “right” instead of `Link` .
- Button should be same as this [button](https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/01be317bc4bbf19461c9eb43a5304f732717305d/files/f58fa9b1-6361-4d06-a6c7-d57f8c9d715d/layers/BF5B326D-31FD-4D85-8D6A-3D81FB59944C?mode=design&sha=01be317bc4bbf19461c9eb43a5304f732717305d)

### How to test

- [ShowMore story](https://picasso.toptal.net/fx-2180-showmore-use-button/?path=/story/components-showmore--showmore)

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Review by OleksandrNechai — APPROVED** [review]
We might have a problem approving with Milos. Looks like they have changed in the approach with underline decoration for the action button. There should be no such decoration. 
If he rises this - just remove decoration in this same PR. It does not seem to be very complicated atm.

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso/src/Button/styles.ts [hunk 1]
```diff
@@ -52,7 +52,8 @@ export default (theme: Theme) => {
       borderRadius: sizes.borderRadius.small,
       border: `solid ${sizes.borderWidth} ${palette.grey.light2}`,
       fontSize: '1rem',
-      transition: `all ${transitions.duration.short}ms ${transitions.easing.easeOut}`,
```
Comment (LashaJini) [inline packages/picasso/src/Button/styles.ts:55]: There is explicit `transitionProperty` after this line, so using `all` in this case, is not necessary. Also, transition didn't look good when hovering on button which shows underline of text.

simplescreenrecorder-2021-11-08_15 04 31

File: packages/picasso/src/ShowMore/ShowMore.tsx [hunk 2]
```diff
@@ -60,23 +60,25 @@ export const ShowMore = forwardRef<HTMLSpanElement, Props>(function ShowMore(
...
           onClick={() => {
             setShownMore(!shownMore)
             onToggle()
           }}
           className={classes.toggleText}
-          variant='action'
+          iconPosition='right'
         >
-          {shownMore ? lessText : moreText}
+          <div className={classes.textWrapper}>
```
Comment (LashaJini) [inline packages/picasso/src/ShowMore/ShowMore.tsx:71]: Currently icon had `margin-left` and `rotation` applied together, which cause something like this:

margin-left-transform

Yes, we could instead applied different margin, but I think it is not a good developer experience and instead I've wrapper text in `div` element and applied right margin there.

File: packages/picasso/src/Button/styles.ts (hunk 1 above)
Comment (TomasSlama) [reply packages/picasso/src/Button/styles.ts:55]: Did you check, if it didn't affect other variants of Button?

File: packages/picasso/src/ShowMore/ShowMore.tsx (hunk 2 above)
Comment (TomasSlama) [reply packages/picasso/src/ShowMore/ShowMore.tsx:71]: wouldn't `span` make more sense? `div` is block element, it seems weird to me to have block element inside button

File: packages/picasso/src/ShowMore/styles.ts [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
   createStyles({
     toggleText: {
       display: 'flex',
       alignItems: 'center',
-      marginTop: '0.5rem'
+      marginTop: '0.5rem',
+      transition: 'none'
+    },
+    textWrapper: {
+      marginRight: rem('8px'),
```
Comment (TomasSlama) [inline packages/picasso/src/ShowMore/styles.ts:13]: on line 9 we already use `0.5rem`, I don't think we need to use `rem()` here. It should be used only when weird spacings need to be converted to rem, for example, 11px, 7px, ...

File: packages/picasso/src/ShowMore/ShowMore.tsx (hunk 2 above)
Comment (TomasSlama) [reply packages/picasso/src/ShowMore/ShowMore.tsx:71]: also when I check examples of `Button.Action`

```
<Button.Action icon={<Link16 />} iconPosition='right'>
  Default
</Button.Action>
```

Is it necessary to use it differently?

File: packages/picasso/src/Button/styles.ts (hunk 1 above)
Comment (LashaJini) [reply packages/picasso/src/Button/styles.ts:55]: It didn't affect. If I didn't miss anything, other buttons animations only include border, color and background transitions.

File: packages/picasso/src/ShowMore/styles.ts (hunk 3 above)
Comment (OleksandrNechai) [reply packages/picasso/src/ShowMore/styles.ts:13]: still not fixed looks like :smile:

**Review by OleksandrNechai — APPROVED** [review]
LGTM :+1:

File: packages/picasso/src/ShowMore/styles.ts [hunk 4]
```diff
@@ -5,15 +5,17 @@ export default ({ palette }: Theme) =>
...
-      marginTop: '0.5rem'
+      marginTop: '0.5rem',
+      transition: 'none',
+      fontWeight: 'normal'
     },
     iconWrapper: {
-      transform: 'rotate(90deg)',
-      fontSize: '0.6em',
-      marginLeft: '1em'
+      fontSize: '1rem !important',
```
Comment (LashaJini) [inline packages/picasso/src/ShowMore/styles.ts:13]: This is necessary to override child's `font-size`, which already has `!important` applied to it. Without this, other components would require huge refactoring.

**Comment by LashaJini** [comment]
@toptal-bot run all

**Pull request merged**